package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -120,32 +117,52 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
 
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
-
-        attrs = [
-            Atomids(np.array(atomids, dtype=np.int32)),
-            Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
-            Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
-            ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
-            Resids(resids),
-            Resnums(resnums),
-            Resnames(resnames),
-            Segids(segids),
-        ]
-
-        top = Topology(
-            n_atoms,
-            n_residues,
-            n_segments,
-            attrs=attrs,
-            atom_resindex=atom_resindex,
-            residue_segindex=residue_segindex,
-        )
+        try:
```

Comment:
Any reason for using a `try/except` here instead of an `if/else` based on the presence of `element` as an attribute? Would the latter not be more readable / lead to fewer duplicated lines?

Reply:
I would also wonder if it's possible for only some elements in the `omm_topology` to be present, much like a PDB file. If that's the case, I'd want my Universe to behave the same as from a PDB file: contain specified elements, with empty strings for the others.

Link to the relevant PDB code: https://github.com/MDAnalysis/mdanalysis/blob/d593993eb5b3ab06ecc3562cb7572605544f77d0/package/MDAnalysis/topology/PDBParser.py#L311-L336